examples/bond/assets/buyback-lsig.py

```diff
@@ -0,0 +1,50 @@
+from pyteal import *
+
+def buyback_lsig():
+    # check no rekeying, close remainder to, asset close to
+    common_fields = And(
+        Gtxn[0].rekey_to() == Global.zero_address(),
+        Gtxn[1].rekey_to() == Global.zero_address(),
+        Gtxn[2].rekey_to() == Global.zero_address(),
+        Gtxn[0].close_remainder_to() == Global.zero_address(),
+        Gtxn[1].close_remainder_to() == Global.zero_address(),
+        Gtxn[2].close_remainder_to() == Global.zero_address(),
+        Gtxn[1].asset_close_to() == Global.zero_address(),
+    )
+
+    # verify that buyer deposits required algos(Not we are checking this in stateful contract)
+    # Here we are verifying if the call is made to bond-dapp stateful contract
+    verify_tx = And(
+        # verify TMPL_BOND is being transferred
+        Gtxn[0].type_enum() == TxnType.AssetTransfer,
+        Gtxn[0].xfer_asset() == Tmpl.Int("TMPL_BOND"),
```

Comment:
I think we can because buyback lsig is only specific to a particular TMPL_BOND, if we change it with epoch address of buyback lsig will be changed , therefore it will be considered different contract

Reply:
for each epoch there will be different buyback address if creator wants to do that, else i think it is meant for last bond token.

Reply:
right, I initially I thought this is `lsig` code (didn't read the buyback prefix) and then I realized it's buyback but forgot to remove the comment.